### PR TITLE
Add Katalyst Core

### PR DESCRIPTION
**Pull request by aedans**
Replaces the arrow-recursion module with [katalyst](https://github.com/aedans/Katalyst) 0.7.0.

### CODE REVIEW

File: modules/recursion-schemes/arrow-recursion/src/test/kotlin/arrow.recursion/laws/RecursiveLaws.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package arrow.recursion.laws
+
+import arrow.recursion.typeclasses.Corecursive
+import arrow.recursion.typeclasses.Recursive
+import arrow.test.laws.Law
+import io.kotlintest.matchers.shouldEqual
+
+object RecursiveLaws {
```
Comment (pakoito) [inline modules/recursion-schemes/arrow-recursion/src/test/kotlin/arrow.recursion/laws/RecursiveLaws.kt]: We have all laws in the `arrow-laws` module which should depend on all `arrow` libraries, if you prefer to have them there!

Comment (aedans) [reply modules/recursion-schemes/arrow-recursion/src/test/kotlin/arrow.recursion/laws/RecursiveLaws.kt]: Makes sense, I'll move them there.

Comment (pakoito) [reply modules/recursion-schemes/arrow-recursion/src/test/kotlin/arrow.recursion/laws/RecursiveLaws.kt]: `forAll(intGen)` can become `forFew`, the rest you can move to the Generators file :D

**Comment by aedans** [comment]
@raulraja Thanks, I'm happy to have it merged! And I think we should announce it once the docs are finished (which I'll be working on soon) so that Arrow users can look at tutorials in Kotlin rather than having to google tutorials in Haskell.

**Pull request merged**